test_dll.py

```diff
@@ -0,0 +1,220 @@
+# Test file for dll.py
+# Authors Mark, Efrain, Henry
+
+import pytest  # used for exception testing
+import dll
+
+
+@pytest.fixture(scope='function')
+def empty_dll():
+    """Fixture generates empty doubly linked list for testing."""
+    return dll.Dll()
+
+
+@pytest.fixture(scope='function')
+def populated_dll():
+    """Fixture generates a populated doubly linked list."""
+    populated = dll.Dll()
+    populated.append('first')
+    populated.append('second')
+    populated.append('third')
+    populated.append('fourth')
+    populated.append(5)
+    populated.append(6)
+    return populated
+
+
+def test_init():
+    """Test doubly link list class constructor."""
+
+    a = dll.Dll()
+    assert a.head is None
+    assert a.tail is None
+
+
+def test_insert(empty_dll, populated_dll):
+    """Test insert by adding node to empty and populated dll
+    """
```

Comment:
There are way way way too many asserts in these functions - they should be split up into more discrete aspects of each function, so you know exactly what works and what doesn't if a test fails.